var/spack/repos/builtin/packages/idl/package.py

```diff
@@ -0,0 +1,44 @@
+from spack import *
+import shutil
+import pathlib
+import os
+
+
+class Idl(Package):
+    """IDL Software: Interactive Data Visulation.
+    
+    Note: IDL is a licensed software. You will also need an existing
+    downloaded tarball of IDL in your current directory or in a 
+    spack mirror in order to install."""
+
+    homepage = "https://www.harrisgeospatial.com/Software-Technology/IDL"
+    idl_mirror_path = 'idl/idl-8.7.tar.gz'
+    manual_download = True
+
+    maintainers = ['francinelapid']
+
+    #version('8.7')
+
+    license_required = True
+
+    def url_for_version(self, version):
+        return "file://{0}/idl{1}-linux.tar.gz".format(os.getcwd(), version)
```

Comment:
You should actually be able to do this with a regular `url`, no need for `url_for_version`